go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -353,23 +385,44 @@ func (cm *ClusterManager) addNode(node *kapi.Node) error {
 					klog.Warningf("Error releasing subnet %v on node %s", allocatedSubnet, node.Name)
 				}
 			}
+			cm.removeNodeId(node.Name, allocatedNodeId)
 		}
 	}()
 
+	allocatedNodeId, err = cm.allocateNodeId(node)
+	if err != nil {
+		return err
+	}
+
 	// Set the HostSubnet annotation on the node object to signal
 	// to nodes that their logical infrastructure is set up and they can
 	// proceed with their initialization
-	nodeSubnetAnnotations, err := util.CreateNodeHostSubnetAnnotation(hostSubnets)
+	nodeAnnotations, err := util.CreateNodeHostSubnetAnnotation(hostSubnets)
 	if err != nil {
 		return fmt.Errorf("failed to marshal node %q annotation for subnet %s",
 			node.Name, util.JoinIPNets(hostSubnets, ","))
 	}
 
+	// Add the node id annotation.
+	nodeAnnotations[util.OvnNodeId] = strconv.Itoa(allocatedNodeId)
+
+	// Generate v4 and v6 transit switch port IPs for the node.
```

Comment:
So, we actually do more than what the commit description suggests.  We also use the node ID to allocate transit switch IPs.  But, at least at this point in the commit history, there's no such thing as a transit switch.

Should we amend the commit description?  Or should we move the transit switch port IP allocation to commit ` ovnkube-master: Create interconnect resources.`?

Reply:
Since all these commits are part of one PR,  I think its ok to have the transit switch ip allocation in this commit.

I've updated the commit message accordingly.